quickwit/quickwit-ingest/src/ingest_v2/mod.rs

```diff
@@ -87,6 +87,29 @@ pub fn get_idle_shard_timeout() -> Duration {
         .unwrap_or(DEFAULT_IDLE_SHARD_TIMEOUT)
 }
 
+const INGEST_ROUTER_BUFFER_SIZE_ENV_KEY: &str = "QW_INGEST_ROUTER_BUFFER_SIZE_BYTES";
+
+const DEFAULT_INGEST_ROUTER_BUFFER_SIZE: u64 = if cfg!(test) { 8 } else { 256 }; // 256 MiB
+
+pub(crate) fn get_ingest_router_buffer_size() -> ByteSize {
+    env::var(INGEST_ROUTER_BUFFER_SIZE_ENV_KEY)
+        .ok()
+        .and_then(|buffer_size_bytes_str| {
+            if let Ok(buffer_size_bytes) = buffer_size_bytes_str.parse::<u64>() {
+                let buffer_size = ByteSize::b(buffer_size_bytes);
+                info!("overriding ingest router buffer size to {buffer_size}");
+                Some(buffer_size)
```

Comment:
(if there are no unit, ByteSize conveniently assumes bytes)

```suggestion
            if let Ok(buffer_size) = buffer_size_bytes_str.parse::<ByteSize>() {
                info!("overriding ingest router buffer size to {buffer_size}");
                Some(buffer_size)
```